.github/workflows/tests.yaml

```
@@ -215,6 +215,52 @@ jobs:
           . venv\\Scripts\\activate
           pytest -vv --durations=20 --benchmark-disable tests/
 
+  tests-macos:
+    name: run / ${{ matrix.python-version }} / macOS
+    runs-on: macos-latest
+    timeout-minutes: 25
+    needs: tests-linux
+    strategy:
+      fail-fast: false
+      matrix:
+        # We only run on the latest supported version on Mac
+        python-version: [3.7]
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v3.0.2
+      - name: Set up Python ${{ matrix.python-version }}
+        id: python
+        uses: actions/setup-python@v3.1.2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Generate partial Python venv restore key
+        id: generate-python-key
+        run: >-
+          echo "::set-output name=key::venv-${{ env.CACHE_VERSION }}-${{
+            hashFiles('setup.cfg', 'requirements_test_min.txt')
+          }}"
+      - name: Restore Python virtual environment
+        id: cache-venv
+        uses: actions/cache@v3.0.2
+        with:
+          path: venv
+          key: >-
+            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{
+            steps.generate-python-key.outputs.key }}
+          restore-keys: |
+            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-venv-${{ env.CACHE_VERSION }}-
+      - name: Create Python virtual environment
+        if: steps.cache-venv.outputs.cache-hit != 'true'
+        run: |
+          python -m venv venv
+          . venv/bin/activate
+          python -m pip install -U pip setuptools wheel
+          pip install -U -r requirements_test.txt
+      - name: Run pytest
+        run: |
+          . venv/bin/activate
+          pytest -vv --durations=20 --benchmark-disable tests/
```

```suggestion
          pytest --durations=20 --benchmark-disable tests/
```
Why use `-vv` for pytest? Just my opinion, but I find it rather annoying to have such a long test output. That's something which bugs me frequently when I look at the Linux test output tbh.

---

We did this when we had the frequent issues with the CI failing to see what failed when since we couldn't reproduce the issues locally and had to get some form of consistent data.

Now that those issues have been fixed we can probably remove those flags again.

Same goes for ``--durations``.

---

I can confirm that we never have any timeout for the main branch but we still have issue on the maintenance 2.13 branch so the issue was definitely fixed in between

---

> Same goes for `--durations`.

Just listing the longest running tests can be helpful sometimes. Maybe a limit of 10 is enough?